### PR TITLE
Process registration

### PR DESCRIPTION
**Pull request by jamiander**
Registration is now working to populate the database and user can be retrieved for login.  Checks if user is already registered.  
User was updated and extra footer removed per Heather's comments.  Not sure about the two css files, I don't think that came from me because I did not set up css.

### CODE REVIEW

**Comment by hstrodtman** [comment]
It's not allowing me to register a user. It says "A user with that email already exists". Do I need to set up anything new in MySQL?

File: src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java [hunk 1]
```diff
@@ -44,45 +50,43 @@ private static void setUserInSession(HttpSession session, User user) {
...
+            return "register";
         }
 
+        Optional<User> existingUser = userRepository.findByUsername(registerDTO.getUsername());
 
-
-        Optional<User> existingUser = userRepository.findByUsername(newUser.getUsername());
-
-        if (existingUser != null) {
+        if (existingUser.isPresent()) {
```
Comment (KBeas21) [inline src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java:69]: Did you mean for this to be not present? Just wondering before it was existingUser != null.

File: src/main/java/org/launchcode/water_garden_tour/models/User.java [hunk 2]
```diff
@@ -4,31 +4,33 @@
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
 import javax.persistence.Table;
+import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
 
 @Entity
-@Table(name = "user")
+//@Table(name = "user")
```
Comment (KBeas21) [inline src/main/java/org/launchcode/water_garden_tour/models/User.java:11]: if this is not needed we can just remove this

Comment (KBeas21) [reply src/main/java/org/launchcode/water_garden_tour/models/User.java:11]: another commented out code could be removed if not need. If you are still in a development stage ignore this message

File: src/main/java/org/launchcode/water_garden_tour/models/dto/RegisterDTO.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    public RegisterDTO() {}
+
+    public String getFname() {
+        return fname;
+    }
+
+    public void setFname(String fname) {
+        this.fname = fname;
+    }
```
Comment (KBeas21) [inline src/main/java/org/launchcode/water_garden_tour/models/dto/RegisterDTO.java:35]: no need to do anything but just wanted to say remember we might not need everything to be public here. Sometimes we might not want getters and setters for everything just making you think. =)

File: src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java (hunk 1 above)
Comment (jamiander) [reply src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java:69]: I changed existingUser !=null to existingUser.isPresent.  The second one works.

File: src/main/java/org/launchcode/water_garden_tour/models/User.java (hunk 2 above)
Comment (jamiander) [reply src/main/java/org/launchcode/water_garden_tour/models/User.java:11]: Not sure if we need it yet.

File: src/main/java/org/launchcode/water_garden_tour/models/dto/RegisterDTO.java (hunk 3 above)
Comment (jamiander) [reply src/main/java/org/launchcode/water_garden_tour/models/dto/RegisterDTO.java:35]: Thank you, good point.

**Comment by hstrodtman** [comment]
Looks good! I was able to register a user and log them in after I dropped the MySQL user table. It looks like there is a conflict now with master though. Possibly something from Ryan's additions yesterday.

**Comment by hstrodtman** [comment]
It seems like there are two footers in the fragments file and two style.css files now.  Also, no need to change right away but the username minimum characters is too small for a lot of emails and the role field needs to be "ROLE_USER" for security.

**Comment by scottpenneman** [comment]
Nice work, Jamie! I was able to run it as well. As for the two CSS files it looks like our head fragment points to css/styles.css, so the one that's just hanging out solo in the static folder can be deleted.

**Comment by MulesKC** [comment]
I was able to register a user and login with that user. Form validation all looks to be working as well. Very impressive work here from the Spring Security team! I had lost styling from css when testing - I think since we kept the css file that was in static the link is now broken in our head fragment. No worries, i think that's an easy fix we can add to another PR later. Great work Jamie!

**Review by scottpenneman — APPROVED** [review]

**Review by MulesKC — APPROVED** [review]
looks good!

**Review by KBeas21 — APPROVED** [review]
+1, Good work go ahead and put this into master, please.

File: src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java (hunk 1 above)
Comment (KBeas21) [reply src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java:69]: just wondering, I wanted to make sure you wouldn't get a NullPointerException from this call. It seems that you will not [per the documentation](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#isPresent--) so resolving this conversation.

**Pull request merged**